tests/panels/test_async_panel_compatibility.py

```diff
@@ -0,0 +1,39 @@
+from django.http import HttpResponse
+from django.test import AsyncRequestFactory, RequestFactory, TestCase
+from django.test.utils import override_settings
+
+from debug_toolbar import settings as dt_settings
+from debug_toolbar.toolbar import DebugToolbar
+
+
+def set_custom_toolbar_config():
+    new_settings = dt_settings.get_config().copy()
+    new_settings["DISABLE_PANELS"] = {}
+    return new_settings
+
+
+class PanelAsyncCompatibilityTestCase(TestCase):
+    def setUp(self):
+        self.factory = AsyncRequestFactory()
+        self.request = self.factory.get("/")
+        self.toolbar = None
+
+    @override_settings(DEBUG_TOOLBAR_CONFIG=set_custom_toolbar_config())
+    def test_async_panel_enabling_with_asgi(self):
+        self.toolbar = DebugToolbar(self.request, lambda request: HttpResponse())
+        for panel in self.toolbar.panels:
+            panel.is_async = False
+            self.assertFalse(panel.enabled)
+            panel.is_async = True
+            self.assertTrue(panel.enabled)
```

Comment:
What's the purpose of this test? Are we testing `Panel.enabled` or each of the individual panels' `enabled` property.

If it's the former, it doesn't need to loop through the panels of the toolbar, it should be able to use an instance of `Panel`. If it's the latter, it shouldn't be setting `is_async` as it changes the panel.

Reply:
It's for the latter, testing each panel `enabled` method individually. 

The goal is to test each panel in 4 different situations, with `ASGI`  `is_async` true or false and with `WSGI` `is_async` true or false, given that other conditions for each of the panel holds true for enabling. That is why `set_custom_config` is used to make `ProfilingPanel` and `RedirectPanel` enable as well on top of the settings used for testing.

Reply:
If you're testing the individual panels, the test shouldn't modify the panel. Does that make sense?

Reply:
I think we can just test them by creating `MockPanel` with `is_async = False` and `is_async = True`
rather than testing each panels individually.